python/ffsim/operators/fermi_hubbard.py

```diff
@@ -58,11 +58,19 @@ def fermi_hubbard_1d(
     """
     coeffs: dict[tuple[tuple[bool, bool, int], ...], complex] = {}
 
+    # initialize tunneling keys
```

Comment:
Let's make `coeffs` a [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict). Then, we won't need this block.

Reply:
Thanks, well spotted! I have changed this for the fermi_hubbard_2d PR as well.